charts/backstage/templates/_helpers.tpl

```diff
@@ -21,3 +21,29 @@ Returns a secret name for service to service auth
         {{- include "common.names.fullname" . -}}-auth
     {{- end -}}
 {{- end -}}
+
+{{/*
+Sets the secretKeyRef name for Backstage to the PostgreSQL existing secret if it present
+*/}}
+{{- define "janus-idp.backstage.existingSecret" -}}
+    {{- if .Values.postgresql.auth.existingSecret -}}
+        {{- .Values.postgresql.auth.existingSecret -}}
+    {{- else if .Values.global.postgresql.auth.existingSecret -}}
+        {{- .Values.global.postgresql.auth.existingSecret -}}
+    {{- else -}}
+        {{- printf "%s-%s" .Release.Name "postgresql" -}}
+    {{- end -}}
+{{- end -}}
+
+{{/*
+Sets the secretKeyRef name for PostgreSQL to the PostgreSQL existing secret if it present
+*/}}
+{{- define "janus-idp.postgresql.existingSecret" -}}
+    {{- if .Values.auth.existingSecret -}}
+        {{- .Values.auth.existingSecret -}}
+    {{- else if .Values.global.postgresql.auth.existingSecret -}}
```

Comment:
`.Values.global.postgresql.auth.existingSecret` has preference over `.Values.auth.existingSecret` in the postgres chart.